test/FormCheckSpec.tsx

```diff
@@ -0,0 +1,264 @@
+    element.classList.contains('is-valid').should.be.true;
+  });
+
+  it('should support isInvalid', () => {
+    const { getByTestId } = render(
+      <FormCheck isInvalid data-testid="test-id" />,
+    );
+
+    const element = getByTestId('test-id');
+    element.classList.length.should.equal(2);
+    element.classList.contains('is-invalid').should.be.true;
+  });
+
+  it('should support ref forwarding', () => {
+    let input;
+    class Container extends React.Component {
+      render() {
+        return (
+          <FormCheck
+            ref={(ref) => {
+              input = ref;
+            }}
+          />
+        );
+      }
+    }
+
+    render(<Container />);
+    input.tagName.toLowerCase().should.equal('input');
+  });
+
+  it('should not render bsPrefix if no label is specified', () => {
+    const { container } = render(
+      <FormCheck id="foo" name="foo" value="foo" type="radio" />,
+    );
+
+    container.getElementsByClassName('form-check').length.should.equal(0);
+  });
+
+  it('should support type switch', () => {
+    const { getByTestId, container } = render(
+      <FormCheck
+        type="switch"
+        label="My label"
+        id="switch-id"
+        data-testid="test-id"
+      />,
+    );
+
+    const element = getByTestId('test-id');
+
+    element.parentElement.classList.length.should.equal(2);
+    element.parentElement.classList.contains('form-check').should.be.true;
+    element.parentElement.classList.contains('form-switch').should.be.true;
+
+    element.id.should.equal('switch-id');
+    element.classList.length.should.equal(1);
+    element.classList.contains('form-check-input').should.be.true;
+    element.id.should.equal('switch-id');
+    element.hasAttribute('type').should.be.true;
+    element.getAttribute('type').should.equal('checkbox');
+
+    const labels = container.getElementsByTagName('label');
+    labels.length.should.equal(1);
+    const label = labels[0];
+    label.classList.length.should.equal(1);
+    label.classList.contains('form-check-label').should.be.true;
+    label.hasAttribute('for').should.be.true;
+    label.getAttribute('for').should.equal('switch-id');
+    label.innerText.should.equal('My label');
+  });
+
+  it('should support Switch', () => {
+    const { getByTestId, container } = render(
+      <Switch label="My label" id="switch-id" data-testid="test-id" />,
+    );
+
+    const element = getByTestId('test-id');
+
+    element.parentElement.classList.length.should.equal(2);
+    element.parentElement.classList.contains('form-check').should.be.true;
+    element.parentElement.classList.contains('form-switch').should.be.true;
+
+    element.id.should.equal('switch-id');
+    element.classList.length.should.equal(1);
+    element.classList.contains('form-check-input').should.be.true;
+    element.id.should.equal('switch-id');
+    element.hasAttribute('type').should.be.true;
+    element.getAttribute('type').should.equal('checkbox');
```

Comment:
Don't need hasAttribute since the value is checked on next line anyway

Reply:
Fixed in the commit https://github.com/react-bootstrap/react-bootstrap/pull/6224/commits/023f9b3e2cffbb507001f992d76bb9cfb4243310